frontend/src/lifetime/input.ts

```diff
@@ -113,6 +111,12 @@ export function createInputManager(editor: EditorState, container: HTMLElement,
 	const onPointerMove = (e: PointerEvent): void => {
 		if (!e.buttons) viewportPointerInteractionOngoing = false;
 
+		// Don't redirect pointer movement to the backend if there's no ongoing interaction and it's over a floating menu on top of the canvas
+		// TODO: A better approach is to pass along a boolean to the backend's input preprocessor so it can know if it's being occluded by the GUI.
+		// TODO: This would allow it to properly decide to act on removing hover focus from something that was hovered in the canvas before moving over the GUI.
```

Comment:
Thoughts on this? I don't have any experience working with the Input Preprocessor system.

Reply:
This means that we just ignore the events in the rust code rather than here in the js code?

Reply:
The problem was that mouse movement *through* a dialog (or other floating menus, like popovers and dropdowns) was being picked up by the backend, causing for example the hovered shapes to show their blue path outlines.

This was how I went about hackily fixing it from the JS side, but it isn't perfect-- for example, hovering over a shape causes its outline to show blue and then this gets stuck in that state (since no subsequent mousemove events are sent to the backend) when it hovers over a floating menu like a dropdown menu, and it only updates (by hiding the outline if the mouse is far enough away) after the mouse is back out from the floating menu.

Filtering these on the JS side so the backend doesn't even hear about these events isn't the right solution. The backend needs to hear about these events, but it needs to be told that the mouse is currently over a part of the UI rather than the viewport so it can act accordingly.

But I don't really know how the IPP code works and I didn't have time to dive in deeper to figure that all out. It's fine as a TODO comment for now, but if you are inclined to try fixing it in this PR or later someday, that'd be awesome (up to you if/when though!).